source/common/http/http1/codec_impl.cc

```diff
@@ -1072,10 +1102,16 @@ Status ServerConnectionImpl::handlePath(RequestHeaderMap& headers, absl::string_
   // request-target. A proxy that forwards such a request MUST generate a
   // new Host field-value based on the received request-target rather than
   // forward the received Host field-value.
+  RETURN_IF_ERROR(checkHeaderEntryAndSendError(Http::HeaderString(header_values.Host),
```

Comment:
I can imagine these errors being pretty hard to debug without trace logs - I think previously the headers would land in the access logs, so you could debug what the invalid host was, where now we're protecting access logs and response filters from bad header data.  I think the loss of debugability is probably worse though so I'd be inclined to add the header and then check validity so we can debug or come up with some other workaround

Reply:
Refactored